AGENTS.md

```diff
@@ -0,0 +1,202 @@
+- `pkg/client/` - Kubernetes client wrappers and utilities
+- `pkg/alert/` - Alert configuration and validation
+- `pkg/metrics/` - Operator metrics and instrumentation
+- `pkg/promqlgen/` - PromQL query generation utilities
+
+### Configuration Management
+Configuration is managed through two ConfigMaps:
+- `cluster-monitoring-config` (in `openshift-monitoring`) - Platform monitoring configuration
+- `user-workload-monitoring-config` (in `openshift-user-workload-monitoring`) - User workload monitoring configuration
+
+Configuration types are defined in `pkg/manifests/types.go` with extensive validation.
+
+## Common Development Commands
+
+### Building
+```bash
+make build              # Build operator and tests-ext binaries
+make operator           # Build only the operator binary
+make operator-no-deps   # Build operator without fetching dependencies
+```
+
+### Jsonnet Manifest Generation
+```bash
+make generate           # Generate all manifests, docs, and test metadata
+make build-jsonnet      # Generate YAML manifests from jsonnet
+make check-assets       # Verify generated assets are up to date
+```
+
+### Local Development
+```bash
+export KUBECONFIG=/path/to/kubeconfig
+make run-local          # Build and run operator locally
+make run-local SWITCH_TO_CMO=false  # Run as current user (e.g., kube:admin)
+```
+
+**Note**: By default, `make run-local` switches to the CMO service account for realistic permissions testing. Use `SWITCH_TO_CMO=false` to maintain your current login (useful when logged in as kube:admin).
+
+### Testing
+```bash
+make test               # Run all tests (unit, rules, e2e, ginkgo)
+make test-unit          # Run unit tests only
+make test-rules         # Validate Prometheus alerting rules
+make test-e2e           # Run end-to-end tests (requires KUBECONFIG)
+make test-ginkgo        # Run ginkgo tests
+
+# Run specific unit test
+go test -v ./pkg/... -run TestHashSecret
+
+# Run specific e2e test
+go test -v -timeout=120m -run TestBodySizeLimit ./test/e2e/ --kubeconfig $KUBECONFIG
+```
+
+### Validation and Verification
+```bash
+make verify             # Run all verification checks
+make format             # Format code (go fmt, jsonnet fmt, shellcheck)
+make golangci-lint      # Run golangci-lint
+make check-rules        # Validate Prometheus rules with promtool
+make check-runbooks     # Verify runbook URLs in alert annotations
+```
+
+### Dependency Management
+```bash
+make vendor             # Update Go module dependencies
+make update             # Update jsonnet dependencies (in jsonnet/jsonnetfile.lock.json)
+```
+
+## Working with Jsonnet
+
+### Updating Jsonnet Dependencies
+When updating vendored jsonnet libraries (e.g., kube-prometheus):
+
+```bash
+cd jsonnet
+# Edit jsonnetfile.json to update version
+jb update
+cd ..
+make generate           # Regenerate all assets
+```
+
+See `Documentation/development.md` for detailed workflow on updating individual jsonnet bundles.
```

Comment:
This reference is probably good enough? No need to spend the additional context.